src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -158,6 +149,14 @@ Transactor::calculateFeePaid(STTx const& tx)
     return tx[sfFee].xrp();
 }
 
+XRPAmount
+Transactor::calculateFee(Application& app, std::uint64_t const baseFee,
+    Fees const& fees, ApplyFlags flags)
+{
+    return scaleFeeLoad(baseFee, app.getFeeTrack(),
```

Comment:
This function is the thinnest of wrappers... At first, I wondered, whether we should just scrap it; it does help document the code by providing a useful context-specific name, although perhaps `minimumFee` would be better still.

With that said, I think we can improve it more: the `baseFee` argument is basically the result of invoking `calculateBaseFee` on the transaction. We currently cache that result and track it from `PreclaimResult` into `ApplyContext`.

I'm wondering if it makes sense to eliminate this, and simply invoke the operation twice. The cost should be minimal; the only expensive operation is loading the `sfSigners` field to calculate its size to adjust the base fee accordingly.

Opinions?

Reply:
I like this idea. `Transactor::caclulateBaseFee` looks inexpensive enough to recalculate. `SetRegularKey::calculateBaseFee` looks more computational expensive than normal, but that should be a rare tx.

Reply:
I just pushed 7548a088b to do this. It simplified quite a few functions.

Reply:
The last commit did it wrong, so I rewrote history and pushed 59c268784. It still requires pre-calculating the base fee and passing it around, but not through quite as much of the process. It still needs to be stored in the `ApplyContext` because of the `static` function override mechanism used, and because the value can potentially change during the processing of the transaction.